dask/dataframe/tests/test_multi.py

```diff
@@ -1152,7 +1152,7 @@ def test_merge_by_index_patterns(how, shuffle_method):
     def fix_index(out, dtype):
         # Workaround pandas bug where output dtype of empty index will be int64
```

Comment:
The comment says "workaround pandas bug". I know it's not your comment, but this is not really a bug, right? it was a choice in pandas? Maybe we can fix the comment too.

Reply:
Adjusted comments, the expected output was actual object on 1.5.x, so fixed that too